### ISSUE TITLE
proto3 fields unnecessarily serialized

### ISSUE THREAD

**Issue by NfNitLoop** [thesayyn/protoc-gen-ts#16]
I found that if I have a message like this:

```proto3
syntax = "proto3";

message Test {
   string foo = 1;
}
```

And I do:
```typescript
// May be "" or non-empty:
let someString = someFieldValue()

new Test({foo:someString}).serialize()
```

... then a string of length 0 gets serialized with the bytes instead of nothing, which I believe should be the expected output for proto3.

> Note that for scalar message fields, once a message is parsed there's no way of telling whether a field was explicitly set to the default value [...] or just not set at all
> [...]
> Also note that if a scalar message field is set to its default, the value will not be serialized on the wire.

from: [proto3 docs](https://developers.google.com/protocol-buffers/docs/proto3)

I can work around this with this pattern for now:
```typescript
let test = new Test()
if (someString) test.foo = someString
```

But it would be nice to have that optimization built in.

**Comment by thesayyn** [comment]
Probably, this is caused by https://github.com/thesayyn/protoc-gen-ts/commit/ad3a1df720ab2aac565cf4bd33ab9559d02274b3
and related to this
https://github.com/thesayyn/protoc-gen-ts/blob/db9e02e6e2f56d2fa55d21f02b681ed93228ccbd/src/index.js#L628

I'll also fix this in this weekend probably.👌

**Comment by thesayyn** [comment]
Fixed by be82cffcabffa6dbfc6260213291932addff5de8.

**Comment by thesayyn** [comment]
this has been fixed in release 0.3.0. let me know how it goes

**Comment by NfNitLoop** [comment]
It works! Thanks! 👍
